salt/states/idem.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+#
+# Author: Tyler Johnson <tjohnson@saltstack.com>
+#
+
+"""
+Idem Support
+============
+
+This state provides access to idem states
+"""
+import logging
+import pathlib
+import sys
+
+import salt.utils.idem as idem
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = "idem"
+
+
+def __virtual__():
+    if sys.version_info < (3, 6):
+        return False, "idem only works on python3.6 and later"
+    if not idem.HAS_POP[0]:
+        return idem.HAS_POP[0], str(idem.HAS_POP[1])
+    return __virtualname__
+
+
+def __init__(opts):
+    if "idem.hub" not in __context__:
+        __context__["idem.hub"] = idem.create_hub()
+
+
+def _get_refs(sources, tree):
+    """
+    Determine where the sls sources are
+    """
+    sls_sources = []
+    SLSs = []
+    if tree:
+        sls_sources.append("file://{}".format(tree))
+    for sls in sources:
+        path = pathlib.Path(sls)
+        if path.is_file():
+            SLSs.append(str(path.stem if path.suffix == ".sls" else path))
```

Comment:
I think you want `(path.parent / path.stem) if path.suffix == ".sls" else path`
You don't want to loose the parent path do you?

Reply:
We do want to lose the parent path in this part.  I actually need to update `path` here to be path.name